Sources/Shared/Extensions/URLExtensions.swift

```diff
@@ -615,4 +615,21 @@ public struct InternalURL {
     }
     return nil
   }
+  
+  public var isWeb3URL: Bool {
+    return web3Component != nil
+  }
+  
+  /// Return the path after "web3t/" in the URI.
+  public var web3Component: String? {
+    let web3Path = "/web3/"
+    guard let url = URL(string: stripAuthorization) else {
```

Comment:
I believe this should be `NSURL(idnString: text) as? URL` cc @Brandon-T

Reply:
updated in [a4cd1a8](https://github.com/brave/brave-ios/pull/6906/commits/a4cd1a858eb2db471ba0d554b8a31267b8cf01fa)